### ISSUE TITLE
FileSystem tests sometimes see wrong System.Runtime.Extensions and fail due to invalid path chars

### ISSUE THREAD

**Issue by nguerrera** [dotnet/corefx#2516]
Locally, I'm seeing failures like the one logged below.

The issue appears to be that the hard-coded TestWithLocalLibraries=true has a build order dependency: if the test builds before System.Runtime.Extensions, it deploys the one from nuget that doesn't like the extended path syntax.

@weshaggard is working on a fix, cc @JeremyKuhne as FYI.

```
[0]: System.ArgumentException: Illegal characters in path.
Parameter name: path
   at System.IO.Path.CheckInvalidPathChars(String path, Boolean checkAdditional)
   at System.IO.Path.GetFullPath(String path)
   at System.IO.PathHelpers.GetFullPathInternal(String path) in C:\src\corefx\src\System.IO.FileSystem\src\System\IO\PathHelpers.Windows.cs:line 46
   at System.IO.DirectoryInfo..ctor(String path) in C:\src\corefx\src\System.IO.FileSystem\src\System\IO\DirectoryInfo.cs:line 21
   at System.IO.FileSystem.Tests.DirectoryInfo_Create.Create(String path) in C:\src\corefx\src\System.IO.FileSystem\tests\DirectoryInfo\Create.cs:line 14
   at System.IO.FileSystem.Tests.Directory_CreateDirectory.<>c__DisplayClass11_0.<ValidExtendedPathWithTrailingSlash>b__0(String component) in C:\src\corefx\src\System.IO.FileSystem\tests\Directory\CreateDirectory.cs:line 128
   at Xunit.Assert.All[T](IEnumerable`1 collection, Action`1 action)
```

**Comment by JeremyKuhne** [comment]
Sorry about that- thanks for catching the issue.

**Comment by nguerrera** [comment]
Fixed by #2518
